graph/src/firehose/endpoints.rs

```diff
@@ -34,15 +34,26 @@ pub struct FirehoseEndpoint {
     pub token: Option<String>,
     pub filters_enabled: bool,
     pub compression_enabled: bool,
-    pub subgraph_limit: usize,
+    pub subgraph_limit: SubgraphLimit,
     channel: Channel,
 }
 
-#[derive(Clone, Debug)]
+// TODO: Find a new home for this type.
+#[derive(Clone, Debug, PartialEq, Ord, Eq, PartialOrd)]
 pub enum SubgraphLimit {
-    Unlimited,
+    Disabled,
     Limit(usize),
-    NoTraffic,
+    Unlimited,
+}
```

Comment:
Do we really need this enum? I don't really think it buys us much above using a plain `usize` (where `Disabled -> 0` and `Unlimited -> usize::MAX`)

Reply:
It's mostly to improve readability since otherwise we'd need to set a value for 0 and some for "not set" in order to differentiate so there would be a few places with Option<usize> where we were not sure of the meaning. Also Unlimited will also get transformed in the different adapters, eg firehose uses 100 max so it gets set to the adapter-specific "maximum"